### PR TITLE
Add support for lelwel language server

### PR DESCRIPTION
**Pull request by 0x2a-42**
https://github.com/0x2a-42/lelwel

### CODE REVIEW

File: lua/lspconfig/lelwel_ls.lua [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Language server for lelwel grammars.
+
+You can install `lelwel-ls` via cargo:
+```sh
+cargo install --features="lsp" lelwel
+```
+]],
+    default_config = {
+      root_dir = "vim's starting directory",
```
Comment (mjlbach) [inline lua/lspconfig/lelwel_ls.lua:22]: The default_config is not accurate, you can just leave it off.

File: lua/lspconfig/lelwel_ls.lua [hunk 2]
```diff
@@ -0,0 +1,25 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+configs.lelwel_ls = {
+  default_config = {
+    cmd = { 'lelwel-ls' },
+    filetypes = { 'llw' },
+    root_dir = util.path.dirname,
```
Comment (mjlbach) [inline lua/lspconfig/lelwel_ls.lua:8]: I would use `.git` as the root dir, see the utility function for finding the git ancestor

File: lua/lspconfig/lelwel_ls.lua (hunk 1 above)
Comment (0x2a-42) [reply lua/lspconfig/lelwel_ls.lua]: OK, I removed it.

File: lua/lspconfig/lelwel_ls.lua (hunk 2 above)
Comment (0x2a-42) [reply lua/lspconfig/lelwel_ls.lua]: Done.

File: lua/lspconfig/lelwel_ls.lua [hunk 3]
```diff
@@ -0,0 +1,24 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+configs.lelwel_ls = {
+  default_config = {
+    cmd = { 'lelwel-ls' },
+    filetypes = { 'llw' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or util.path.dirname(fname)
```
Comment (mjlbach) [inline lua/lspconfig/lelwel_ls.lua:9]: You can remove util.path.dirname now, it's handled automatically.

root_dir = util.find_git_ancestor

**Comment by mjlbach** [comment]
You can test if `single_file_support = true` works for this config and enable in a follow-up PR, I'm merging.

**Pull request merged**